app/connection/ConnectionManager.js

```diff
@@ -110,7 +116,15 @@ class ConnectionManager extends EventEmitter {
         }, timeout);
     }
 
-    async send(data, timeout) {
+    _sendBufferedRequests() {
+        while (this._bufferedRequests.length > 0) {
+            const req = this._bufferedRequests.shift();
+            this.send( req.data , req);
```

Comment:
You changed the method signature to `async send(data, promise, timeout) {`? This doesn't match